eng/pipelines/docindex.yml

```diff
@@ -50,29 +50,16 @@ jobs:
       - template: /eng/common/pipelines/templates/steps/set-daily-docs-branch-name.yml
         parameters:
           DailyBranchVariableName: DailyDocsBranchName
-      # Create a branch to track docs fork repo onboarding files/folders
-      - pwsh: |
-          $ErrorActionPreference = "Continue"
-          # git remote on fork repo
-          $GitUrl = "https://$(azuresdk-github-pat)@github.com/$(DailyDocsRepoOwner)/$(DocRepoName).git"
-          $remoteName = "$(DailyDocsRepoOwner)"
-          Write-Host "git remote add $remoteName $GitUrl"
-          git remote add $remoteName $GitUrl
-          if ($LASTEXITCODE -ne 0)
-          {
-            Write-Error "Unable to add remote LASTEXITCODE=$($LASTEXITCODE), see command output above."
-            exit $LASTEXITCODE
-          }
-          
-          # git fetch on fork repo
-          Write-Host "git fetch $remoteName $(DailyDocsBranchName)"
-          git fetch $remoteName "$(DailyDocsBranchName)"
```

Comment:
@benbp it would be good to understand why adding another remote and then fetching a branch to checkout doesn't work as expected when using sparse checkout. https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1233498&view=logs&j=dc056dfc-c0cf-5958-c8c4-8da4f91a3739&t=d08ed1f2-0ec2-5498-48a9-682643d126ca

Reply:
Based on git man documentation, git fetch do downloading the deltas.
There is no sparse checkout like functionality for git fetch

https://stackoverflow.com/questions/30903987/does-sparse-checkout-affect-git-fetch

Reply:
https://stackoverflow.com/questions/4114887/is-it-possible-to-do-a-sparse-checkout-without-checking-out-the-whole-repository

Reply:
I don't think that is the same situation that we have here. We already do the no-checkout on clone but we should be able to add a remote and then fetch a new set of commits down and checkout just those changes.